### PR TITLE
Don't do empty background sync triggers (#5436)

### PR DESCRIPTION
**Pull request by mathewc**
Addresses https://github.com/Azure/azure-functions-host/issues/5436

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs [hunk 1]
```diff
@@ -99,27 +99,39 @@ public async Task<SyncTriggersResult> TrySyncTriggersAsync(bool checkHash = fals
...
-                string json = await GetSyncTriggersPayload();
+                var payload = await GetSyncTriggersPayload();
+                if (isBackgroundSync && payload.Count == 0)
+                {
+                    // We don't do background sync for empty triggers.
+                    // We've seen error cases where a site temporarily gets into a situation
+                    // where it's site content is empty. Doing the empty sync can cause the app
+                    // to go idle when it shouldn't.
+                    _logger.LogDebug("No functions found. Skipping Sync operation.");
+                    return new SyncTriggersResult
```
Comment (alrod) [inline src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs]: We can `return result;` here.

File: src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs [hunk 2]
```diff
@@ -99,27 +99,39 @@ public async Task<SyncTriggersResult> TrySyncTriggersAsync(bool checkHash = fals
                 await _syncSemaphore.WaitAsync();
 
                 var hashBlob = await GetHashBlobAsync();
-                if (checkHash && hashBlob == null)
+                if (isBackgroundSync && hashBlob == null)
                 {
-                    // short circuit before doing any work in cases where
-                    // we're asked to check/update hash but don't have
+                    // short circuit before doing any work in backgorund sync
```
Comment (alrod) [inline src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs]: Do we need to add debug log in this case?

Comment (mathewc) [reply src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs]: It's not really an error case - people can run functions without a storage account configured. So I'd rather not have the noise logs.

File: src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs (hunk 1 above)
Comment (mathewc) [reply src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs]: fixed

**Pull request merged**